proto/cosmos/staking/v1beta1/staking.proto

```diff
@@ -320,6 +320,10 @@ message Params {
     (amino.dont_omitempty) = true,
     (cosmos_proto.scalar)  = "cosmos.Dec"
   ];
+  // max_cons_pubkey_rotations is maximum number of consensus pubkey rotations
+  uint64 max_cons_pubkey_rotations = 7;
+  // cons_pubkey_rotation_fee is fee to be spent when rotating validator's consensus pubkey
+  cosmos.base.v1beta1.Coin cons_pubkey_rotation_fee = 8 [(gogoproto.nullable) = false];
```

Comment:
The test failure is saying

```
Diff:
--- Expected
+++ Actual
@@ -1 +1 @@
-{"type":"cosmos-sdk/x/staking/MsgUpdateParams","value":{"params":{"unbonding_time":"0","min_commission_rate":"0","cons_pubkey_rotation_fee":{"amount":"0"}}}}
+{"type":"cosmos-sdk/x/staking/MsgUpdateParams","value":{"params":{"unbonding_time":"0","min_commission_rate":"0"}}}
```

so it looks to me that you need to (at least) add `amino.dont_omitempty` here to preserve parity between the legacy and new aminojson encoder.

Reply:
As an aside, this is the type of issue we're thinking an implementation of https://github.com/cosmos/cosmos-sdk/issues/14682 might be able to catch.

Reply:
> so it looks to me that you need to (at least) add amino.dont_omitempty here to preserve parity between the legacy and new aminojson encoder.

This change didn't help to fix the test @kocubinski